app/src/main/java/org/simple/clinic/drugs/EditMedicinesUiRenderer.kt

```diff
@@ -10,14 +10,43 @@ class EditMedicinesUiRenderer(private val ui: EditMedicinesUi) : ViewRenderer<Ed
 
   override fun render(model: EditMedicinesModel) {
     if (model.prescribedDrugs != null && model.protocolDrugs != null)
-      renderPrescribedProtocolDrugs(model.prescribedDrugs, model.protocolDrugs)
+      renderPrescribedProtocolDrugs(model, model.prescribedDrugs, model.protocolDrugs)
   }
 
-  private fun renderPrescribedProtocolDrugs(prescribedDrugs: List<PrescribedDrug>, protocolDrugs: List<ProtocolDrugAndDosages>) {
+  private fun renderPrescribedProtocolDrugs(
+      model: EditMedicinesModel,
+      prescribedDrugs: List<PrescribedDrug>,
+      protocolDrugs: List<ProtocolDrugAndDosages>
+  ) {
     val prescribedProtocolDrugs = prescribedDrugs.filter { it.isProtocolDrug }
```

Comment:
We should not be using `isProtocolDrug` property on `PrescribedDrug` anymore since it is unreliable. The only thing we'll use it for is for reporting on the dashboard. The app should check using the `ProtocolDrugAndDosage` check we introduced earlier.

With that in mind, I think we can use `partition` to split the list into protocol and custom drugs easily:

```kotlin
val (prescribedProtocolDrugs, prescribedCustomDrugs) = prescribedDrugs.partition(model::isProtocolDrug)
```

Reply:
I had no idea Kotlin has a method called `partition`, will use that.

Reply:
Done